crates/services/producer/src/block_producer.rs

```diff
@@ -171,23 +160,12 @@ where
         height: BlockHeight,
         block_time: Option<Tai64>,
     ) -> anyhow::Result<PartialBlockHeader> {
-        let previous_block_info = self.previous_block_info(height)?;
-        let new_da_height = self
-            .select_new_da_height(previous_block_info.da_height)
-            .await?;
+        let mut block_header = self._new_header(height, block_time)?;
+        let new_da_height = self.select_new_da_height(block_header.da_height).await?;
```

Comment:
So the delay for dry runs was caused by awaiting on the relayer?

Reply:
Yes, it caused 5 seconds delays and queuing requests. After the import of the eth block, it seems that the `synced` became `None` and waited for an actualization after `sync_minimum_duration`.

<img width="655" alt="image" src="https://user-images.githubusercontent.com/18346821/233815590-9256807e-8fe9-44b4-977f-a2b7706e06f7.png">

I'm not sure how it is better to fix it because we may need to really request some blocks again that we could miss while we loaded the previous one. So I fixed that on the `dry_run` side

Reply:
It should return whatever the last synced state was, and only block temporarily while the state is being updated.

Reply:
we might need to add a separate cached state that is always available to poll, and just uses AtomicU64 to update the block height. We should really only wait if the relayer is really far behind the da layer.

Reply:
https://github.com/FuelLabs/fuel-core/pull/1151/commits/46853f3af368df2b504c93395a747c24923bcd31